operators/advanced_network/python/adv_network_common_pybind.cpp

```diff
@@ -17,109 +17,316 @@
+  m.def("get_segment_packet_ptr",
+        py::overload_cast<BurstParams*, int, int>(&get_segment_packet_ptr),
+        "Get packet pointer for one segment",
+        "burst"_a,
+        "seg"_a,
+        "idx"_a);
+  m.def("get_packet_ptr",
+        py::overload_cast<BurstParams*, int>(&get_packet_ptr),
+        "Get packet pointer",
+        "burst"_a,
+        "idx"_a);
+
+  // TX burst functions
   m.def("tx_burst_available",
         py::overload_cast<BurstParams*>(&is_tx_burst_available),
-        "Return true if a TX burst is available for use");
+        "Return true if a TX burst is available for use",
+        "burst"_a);
   m.def("get_tx_packet_burst",
         py::overload_cast<BurstParams*>(&get_tx_packet_burst),
-        "Get TX packet burst");
-  m.def("shutdown", (&shutdown), "Shut down the advanced_network manager");
-  m.def("print_stats", (&print_stats), "Print statistics from the advanced_network manager");
-  // m.def("set_cpu_udp_payload",
-  //     [](BurstParams *burst, int idx, long int data, int len) {
-  //             return set_cpu_udp_payload(burst, idx,
-  //                     reinterpret_cast<void*>(data), len); },
-  //             "Set UDP header parameters and copy payload");
-  // m.def("set_cpu_udp_payload",
-  //     [](std::shared_ptr<BurstParams> burst, int idx, long int data, int len) {
-  //         return set_cpu_udp_payload(burst, idx,
-  //              reinterpret_cast<void*>(data), len); },
-  //         "Set UDP header parameters and copy payload");
+        "Get TX packet burst",
+        "burst"_a);
+  m.def("send_tx_burst",
+        py::overload_cast<BurstParams*>(&send_tx_burst),
+        "Send a TX burst",
+        "burst"_a);
+
+  // RX burst functions - all overloads
+  m.def(
+      "get_rx_burst",
+      [](int port, int q) {
+        BurstParams* burst_ptr = nullptr;
+        Status status = get_rx_burst(&burst_ptr, port, q);
+        return py::make_tuple(status, py::cast(burst_ptr, py::return_value_policy::reference));
+      },
+      "Get RX burst for specific port and queue",
+      "port"_a,
+      "q"_a);
+
+  m.def(
+      "get_rx_burst",
+      [](int port) {
+        BurstParams* burst_ptr = nullptr;
+        Status status = get_rx_burst(&burst_ptr, port);
+        return py::make_tuple(status, py::cast(burst_ptr, py::return_value_policy::reference));
+      },
+      "Get RX burst from any queue on specific port",
+      "port"_a);
 
+  m.def(
+      "get_rx_burst",
+      []() {
+        BurstParams* burst_ptr = nullptr;
+        Status status = get_rx_burst(&burst_ptr);
+        return py::make_tuple(status, py::cast(burst_ptr, py::return_value_policy::reference));
+      },
+      "Get RX burst from any queue on any port");
+
+  // Header setting functions
+  m.def("set_eth_header",
+        py::overload_cast<BurstParams*, int, char*>(&set_eth_header),
+        "Set Ethernet header in packet",
+        "burst"_a,
+        "idx"_a,
+        "dst_addr"_a);
```

Comment:
The `char*` here should probably be `const char*` in the underlying C++ code? (i.e. this function uses the char* as input but does not modify it?)

It may be better to use some lambda to pass a Python `str` containing the MAC adddress like this:

```suggestion
m.def("set_eth_header",
      [](BurstParams* burst, int idx, const std::string& dst_addr) -> Status {
        char mac_bytes[6];
        format_eth_addr(mac_bytes, dst_addr);  // Use existing utility function
        return set_eth_header(burst, idx, mac_bytes);
      },
      "Set Ethernet header in packet",
      "burst"_a,
      "idx"_a,
      "dst_addr"_a);
```

or if it expected that bytes would be used instead?
```python
m.def("set_eth_header",
      [](BurstParams* burst, int idx, py::bytes dst_addr) -> Status {
        std::string addr_str = py::cast<std::string>(dst_addr);
        if (addr_str.size() != 6) {
          throw std::invalid_argument("MAC address must be 6 bytes");
        }
        return set_eth_header(burst, idx, const_cast<char*>(addr_str.c_str()));
      },
      "Set Ethernet header in packet",
      "burst"_a,
      "idx"_a,
      "dst_addr"_a);
```